src/pkgcheck/checks/profiles.py

```diff
@@ -411,9 +411,7 @@ def finish(self):
                 continue
             for parent in profile.stack:
                 seen_profile_dirs.update(dir_parents(parent.name))
-                # flag lagging profile EAPIs -- assumes EAPIs are sequentially
-                # numbered which should be the case for the gentoo repo
-                if (self.options.gentoo_repo and str(profile.eapi) < str(parent.eapi)):
+                if profile.eapi != parent.eapi and profile.eapi in parent.eapi.inherits:
```

Comment:
I should have mentioned as well, I'd use the ``is`` operator rather than equality since EAPI objects are immutable, i.e.

```python
if profile.eapi is not parent.eapi
```

IIRC, I don't think the EAPI class even explicitly defines the related equality magic methods, but I guess they might come from auto-magic snakeoil inheritance in some fashion.

Reply:
Ah nevermind, I forgot that python uses the ``is`` operator internally if you don't provide a specific implementation for the __eq__, but personally I'd still consider using it explicitly instead.

Reply:
Sure.